flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -156,6 +159,9 @@ public boolean scaleResource(
 
         autoScalerStateStore.storeConfigChanges(context, configOverrides);
 
+        // Try to clear all delayed scale down requests after scaling.
+        delayedScaleDown.clearAll();
```

Comment:
Hi @gyfora , reply your comment[1] here.

IIUC, the previous logic still exists, if the rescale happens, `delayedScaleDown.clearAll` will be called (rescale includes scale up). It means the first trigger time is reset. 

Not sure whether your question is answered.

[1] https://github.com/apache/flink-kubernetes-operator/pull/875#discussion_r1738062495

Reply:
This logic isn't same with the previous logic.

## One demo of previous logic

- 9:00 scale up (be executed)
- 9:30 scale down (ignore)
- 9:59 scale down (ignore)
- 10:00 scale down (can be executed)

## New logic 

- 9:00 scale up (be executed, clear all first trigger time)
- 9:30 scale down (update the first trigger time to 9:30)
- 10:00 scale down (within interval from 9:30 to 10:30)
- 10:29 scale down (within interval from 9:30 to 10:30)
- 10:30 scale down (can be executed)

Reply:
So if there was a scale down before 9:00 and that set the first trigger time, that doesn't matter as the scale up resets it subsequently.

In that case we should probably rename the method ` detectImmediateScaleDown` to something like `applyScaleDownInterval`